plugins/secure-mode-plugin/src/main/java/ai/djl/serving/plugins/securemode/SecureModeUtils.java

```diff
@@ -0,0 +1,390 @@
+    public static void reconcileSources(String modelDir) throws IOException {
+        List<String> trustedPathList =
+                splitCommaSeparatedString(Utils.getenv(TRUSTED_CHANNELS_ENV_VAR));
+        linkAdditionalRequirementsTxt(trustedPathList, Paths.get(modelDir));
+    }
+
+    /**
+     * Check if a disallowed option is set via environment variables. The disallowed value is
+     * handled on a per-option basis.
+     *
+     * @throws ModelException if a security check fails
+     */
+    private static void checkOptionEnvVars() throws ModelException {
+        List<String> securityControls =
+                splitCommaSeparatedString(Utils.getenv(SECURITY_CONTROLS_ENV_VAR));
+
+        if (securityControls.contains(TRUST_REMOTE_CODE_CONTROL)) {
+            String optionEnvValue = Utils.getenv("OPTION_TRUST_REMOTE_CODE");
+            if (optionEnvValue != null && Boolean.parseBoolean(optionEnvValue.trim())) {
+                throw new ModelException(
+                        "Setting OPTION_TRUST_REMOTE_CODE to True is prohibited in Secure Mode.");
+            }
+        }
+        if (securityControls.contains(CUSTOM_ENTRYPOINT_CONTROL)) {
+            // Check for both env vars that can be used to set entrypoint
+            String optionEnvValue = Utils.getenv("OPTION_ENTRYPOINT");
+            String djlEnvValue = Utils.getenv("DJL_ENTRY_POINT");
+            if ((optionEnvValue != null && optionEnvValue.trim().endsWith(".py"))
+                    || (djlEnvValue != null && djlEnvValue.trim().endsWith(".py"))) {
+                throw new ModelException(
+                        "Setting OPTION_ENTRYPOINT or DJL_ENTRY_POINT to a custom Python script is"
+                                + " prohibited in Secure Mode. It must be set to a built-in handler"
+                                + " provided by LMI (i.e. starting with `djl_python`).");
+            }
+        }
+    }
+
+    /**
+     * Given a list of absolute paths, scan each path with enabled security checks.
+     *
+     * @param pathList list of absolute paths
+     * @throws IOException if there is an error scanning the paths
+     * @throws ModelException if a security check fails
+     */
+    private static void scanPaths(List<String> pathList) throws IOException, ModelException {
+        List<String> securityControls =
+                splitCommaSeparatedString(Utils.getenv(SECURITY_CONTROLS_ENV_VAR));
+        for (String path : pathList) {
+            Path p = Paths.get(path.trim());
+            if (Files.isDirectory(p)) {
+                if (securityControls.contains(REQUIREMENTS_TXT_CONTROL)) {
+                    scanForRequirementsTxt(p);
+                }
+                if (securityControls.contains(PICKLE_FILES_CONTROL)) {
+                    scanForPickle(p);
+                }
+                if (securityControls.contains(TRUST_REMOTE_CODE_CONTROL)) {
+                    scanForDisallowedOption(p, "option.trust_remote_code");
```

Comment:
It is a generalization for potential future functionality, in the future there may be SM scenarios where a `serving.properties` provided by the auxiliary paths is merged with/overrides the main one – the code to do this would needed to be added later